src/hooks/useMutation/index.ts

```diff
@@ -0,0 +1,55 @@
+import { useState } from 'react';
+import { IMutationState, TArgsMutationFunction, TUseMutation } from './useMutation.types';
+import { AxiosError } from 'axios';
+import { api } from 'services';
+
+const useMutation = (): TUseMutation & IMutationState => {
+  const [state, setState] = useState<IMutationState>({
+    data: null,
+    error: null,
+    isError: false,
+    isLoading: false,
+    status: 'idle',
+  });
+
+  const mutate = async ({
+    method = 'post',
+    url,
+    data,
+    onSuccess,
+    onError,
+  }: TArgsMutationFunction): Promise<void> => {
```

Comment:
do NOT use types like this
use like this
const mutation: TMutationFn = useCallback(...)